ssv/proposer.go

```diff
@@ -232,19 +240,29 @@ func (r *ProposerRunner) expectedPreConsensusRootsAndDomain() ([]ssz.HashRoot, p
 
 // expectedPostConsensusRootsAndDomain an INTERNAL function, returns the expected post-consensus roots to sign
 func (r *ProposerRunner) expectedPostConsensusRootsAndDomain() ([]ssz.HashRoot, phase0.DomainType, error) {
+	var root phase0.Root
 	if r.decidedBlindedBlock() {
-		_, data, err := r.GetState().DecidedValue.GetBlindedBlockData()
+		vBlk, err := r.GetState().DecidedValue.GetBlindedBlockData()
 		if err != nil {
 			return nil, phase0.DomainType{}, errors.Wrap(err, "could not get blinded block data")
 		}
-		return []ssz.HashRoot{data}, types.DomainProposer, nil
+		root, err = vBlk.Root()
+		if err != nil {
+			return nil, phase0.DomainType{}, errors.Wrap(err, "could not get blinded block root")
+		}
+	} else {
+		vBlk, err := r.GetState().DecidedValue.GetBlockData()
+		if err != nil {
+			return nil, phase0.DomainType{}, errors.Wrap(err, "could not get block data")
+		}
+		root, err = vBlk.Root()
+		if err != nil {
+			return nil, phase0.DomainType{}, errors.Wrap(err, "could not get block root")
+		}
 	}
 
-	_, data, err := r.GetState().DecidedValue.GetBlockData()
-	if err != nil {
-		return nil, phase0.DomainType{}, errors.Wrap(err, "could not get block data")
-	}
-	return []ssz.HashRoot{data}, types.DomainProposer, nil
+	hashRoot := types.SSZ32Bytes(root)
```

Comment:
@GalRogozinski notice you returned the root instead of object here right?

Reply:
you mean return `vBlk` instead of `[]ssz.HashRoot{hashRoot}`. That's even more refactoring considering that the current function name is `expectedPostConsensusRootsAndDomain()`.

I wouldn't spend farther time refactoring this code. Given that the current way isn't bad in my view and there are more pressing issues.